experiments/kompanion/cmd/summary/summary.go

```diff
@@ -0,0 +1,398 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package summary
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/experiments/kompanion/pkg/utils"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	examples = `
+	# Summarize Config Connector resources across all namespaces, excludes \"kube\" namespaces by default
+	kompanion summary
+	`
+)
+
+func BuildSummaryCmd() *cobra.Command {
+	opts := NewSummaryOptions()
+	cmd := &cobra.Command{
+		Use:     "summary",
+		Short:   "summarize Config Connector resources",
+		Example: examples,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return RunSummarize(cmd.Context(), opts)
+		},
+		Args: cobra.ExactArgs(0),
+	}
+
+	flags := cmd.Flags()
+	opts.AddFlags(flags)
+
+	return cmd
+}
+
+type Result struct {
```

Comment:
I think this is fine for now but adding some types around the Map[gk, Map[status, count]]] might make this easier to read.

Reply:
Is it possible to represent this as a 3-uple instead? [<NS,GK,status> => count] ? would this help simplify increment?

Reply:
Its possible but that then makes other things harder. If you want to count objects in a namespace you now have to walk the entire thrupple map and filter then accrue, rather than just walking the given namespace map and accruing.